ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/image/ImageComponentView.kt

```diff
@@ -0,0 +1,138 @@
+package com.revenuecat.purchases.ui.revenuecatui.components.image
+
+import android.graphics.Color
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.draw.drawWithCache
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.revenuecat.purchases.paywalls.components.properties.ColorInfo
+import com.revenuecat.purchases.paywalls.components.properties.Size
+import com.revenuecat.purchases.paywalls.components.properties.SizeConstraint
+import com.revenuecat.purchases.ui.revenuecatui.components.modifier.size
+import com.revenuecat.purchases.ui.revenuecatui.components.style.ImageComponentStyle
+import com.revenuecat.purchases.ui.revenuecatui.composables.RemoteImage
+import com.revenuecat.purchases.ui.revenuecatui.extensions.applyIfNotNull
+import java.net.URL
+import androidx.compose.ui.graphics.Color as ComposeColor
+
+@Suppress("SpreadOperator", "LongParameterList")
+@JvmSynthetic
+@Composable
+internal fun ImageComponentView(
+    style: ImageComponentStyle,
+    modifier: Modifier = Modifier,
+) {
+    if (style.visible) {
+        Box(modifier = modifier.applyIfNotNull(style.shape) { clip(it) }) {
```

Comment:
Had to apply the clipping in an outer box... If I applied it to the image itself, it didn't work together with the gradient overlays.

Reply:
You can change the `drawWithCache` modifier to draw an `Outline` instead of a `Rect`. That way we don't need the `Box`. Might be nice to turn it into a custom modifier (for readability and reusability)? Something like: 

```kotlin
@JvmSynthetic
@Stable
internal fun Modifier.overlay(
    color: ColorStyle,
    shape: Shape = RectangleShape,
): Modifier = this then drawWithCache {
    val outline = shape.createOutline(size, layoutDirection, this)

    onDrawWithContent {
        drawContent()
        when (color) {
            is ColorStyle.Solid -> drawOutline(outline, color.color)
            is ColorStyle.Gradient -> drawOutline(outline, color.brush)
        }
    }
}
```

Which can then be used as: 

```kotlin
.applyIfNotNull(style.colorStyle) { colorStyle ->
    overlay(colorStyle, style.shape ?: RectangleShape)
}
// This is now needed, as the Box is no longer clipping the image itself.
.applyIfNotNull(style.shape) { clip(it) }
```